firmware/lib/Axis/src/Axis.cpp

```diff
@@ -0,0 +1,264 @@
+#include "Axis.h"
+
+// -------------------------------- X AXIS SETUP -------------------------------- //
+Axis setup_axis_x()
+{  
+    // Set up the timer interrupt
+    // TC1 channel 0, the IRQ for that channel and the desired frequency
+    pinMode(ENCODER_PIN_A_X, INPUT);
+    pinMode(ENCODER_PIN_B_X, INPUT);
+    Encoder encoder_x = {ENCODER_PIN_A_X, 10, 0};
+
+    pinMode(LIMIT_SW_HOME_PIN_X, INPUT_PULLUP);
+    LimitSwitch ls_home = {LIMIT_SW_HOME_PIN_X, digitalRead(LIMIT_SW_HOME_PIN_X)};
+
+    pinMode(LIMIT_SW_FAR_PIN_X, INPUT_PULLUP);
+    LimitSwitch ls_far_from_home = {LIMIT_SW_FAR_PIN_X, digitalRead(LIMIT_SW_FAR_PIN_X)};
+    
+    pinMode(DIR_PIN_X, OUTPUT);
+    pinMode(STEP_PIN_X, OUTPUT);
+    Axis axis_x_temp = {DIR_PIN_X, STEP_PIN_X, ACCEL_X, VEL_MAX_X, VEL_MIN_X, 
+                        {0, 0, 0}, 0, encoder_x, ls_home, ls_far_from_home,
+                        TC1, 0, 1, TC3_IRQn, TC4_IRQn};
+    return axis_x_temp;
```

Comment:
This method is basically identical to the y setup one. Can we do something like declare a:
```c
typedef struct {
    uint32_t encoderPinA, encoderPinB;
    uint32_t limitSwHomePin;
    etc.
} AxisConfig;
```
Then have one method that you pass a `AxisConfig *` to

Reply:
Cool, will look into making it more concise